server/plugin/plugin.go

```diff
@@ -133,7 +133,17 @@ func (p *Plugin) OnConfigurationChange() (err error) {
 		e.Dependencies.Poster = e.bot
 		e.Dependencies.Store = store.NewPluginStore(p.API, e.bot)
 		e.Dependencies.IsAuthorizedAdmin = p.IsAuthorizedAdmin
-		e.Dependencies.SettingsPanel = mscalendar.NewSettingsPanel(e.bot, e.Dependencies.Store, e.Dependencies.Store, "/settings", pluginURL)
+		e.Dependencies.SettingsPanel = mscalendar.NewSettingsPanel(
+			e.bot,
+			e.Dependencies.Store,
+			e.Dependencies.Store,
+			"/settings",
+			pluginURL,
+			func(userID string) string {
```

Comment:
I think we should return `(string, error)`. As I currently understand, if we can't get their timezone, then we can't properly set the settings.